### PR TITLE
Upgrade go to 1.18.4

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

Security fixes have been released:

https://groups.google.com/g/golang-announce/c/nqrv9fbR0zE

But there's also no reason to being stuck on go 1.17 when go 1.19 is
coming out next month, so bumping to go 1.18.4

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/grafana/mimir
 
-go 1.17
+go 1.18
```
Comment (aknuds1) [inline go.mod]: Is there any need to require Go 1.18 for the Mimir module though? Should think that's independent of upgrading to Go 1.18 for ourselves?

**Review by aknuds1 — COMMENTED** [review]
Generally looks good, just questioning the bumping of the `go` directive in go.mod.

File: go.mod (hunk 1 above)
Comment (colega) [reply go.mod]: Reverted: https://github.com/grafana/mimir/pull/2400/commits/348a8a7230503e798d5d572e47555a82e8ae8ef0

**Review by aknuds1 — APPROVED** [review]
LGTM, AFAIK there shouldn't be any problems stemming from bumping our builds from 1.17 to 1.18.

**Review by pstibrany — APPROVED** [review]

**Review by pracucci — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (zalegrala) [reply go.mod]: There is a new package in 1.18 `net/netip` that I'd like to make use of in dskit.  The thought was that if all the projects were updated to 1.18 we could bump the version in dskit also.

Comment (colega) [reply go.mod]: See: https://github.com/grafana/mimir/pull/2550

**Pull request merged**